src/osdc/Objecter.h

```diff
@@ -2058,6 +2058,9 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   }
   void ms_fast_dispatch(Message *m) {
     ms_dispatch(m);
+    if (!ms_dispatch(m)) {
```

Comment:
@smithfarm this backport is buggy.

we should call `ms_dispatch()` for only one time. so this patch should instead look like
```diff
diff --git a/src/osdc/Objecter.h b/src/osdc/Objecter.h
index 8167306dd8..2955a071e7 100644
--- a/src/osdc/Objecter.h
+++ b/src/osdc/Objecter.h
@@ -2057,7 +2057,9 @@ private:
     }
   }
   void ms_fast_dispatch(Message *m) {
-    ms_dispatch(m);
+    if (!ms_dispatch(m)) {
+      m->put();
+    }
   }

   void handle_osd_op_reply(class MOSDOpReply *m);
```

Reply:
@tchaikov ACK I overlooked that line when resolving the cherry-pick conflict. Fixed now. Thanks a lot!